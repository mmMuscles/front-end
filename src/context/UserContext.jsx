import { createContext, useContext, useMemo, useState } from 'react';
import { supabase } from '../services/supabaseClient';

const UserContext = createContext();

const UserProvider = ({ children }) => {
  const session = supabase.auth.session()
  const [profile, setProfile] = useState({})
  const [user, setUser] = useState(
    session ? { id: session.user.id, email: session.user.email } : {}
  );
  
  const value = useMemo(() => ({ user, setUser, profile, setProfile}), [user, profile]);

  return <UserContext.Provider value={value}>{children}</UserContext.Provider>;
};

const useUser = () => {
  const context = useContext(UserContext);
  if (context === undefined) {
    throw new Error('useUser must be used within a UserProvider');
  }
  return context;
};

export { UserProvider, useUser };
