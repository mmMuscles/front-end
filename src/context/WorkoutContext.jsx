import { useContext, createContext, useState } from "react";



const workoutCtx = createContext();


export const WorkoutProvider = ({ children }) => {
    const [workouts, setWorkouts] = useState([])

return <workoutCtx.Provider value={{workouts, setWorkouts}}>{children}</workoutCtx.Provider>
}



export const useWorkout = () => {
    const context = useContext(workoutCtx)

    if(context === undefined){
        throw new Error('Invalid Provider')
    }

    return context;
}
