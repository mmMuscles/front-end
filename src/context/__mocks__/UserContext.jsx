import { createContext, useContext, useMemo, useState } from "react";

const UserContext = createContext();

const UserProvider = ({ children, mockUser }) => {
  const [profile, setProfile] = useState({});
  const [user, setUser] = useState(
    mockUser ? { id: mockUser.id, email: mockUser.email } : {}
  );

  const value = useMemo(
    () => ({ user, setUser, profile, setProfile }),
    [user, profile]
  );

  return <UserContext.Provider value={value}>{children}</UserContext.Provider>;
};

const useUser = () => {
  const context = useContext(UserContext);
  if (context === undefined) {
    throw new Error("useUser must be used within a UserProvider");
  }
  return context;
};

export { UserProvider, useUser };
