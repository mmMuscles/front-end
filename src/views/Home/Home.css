h1 {
    background-image: url('../../assets/pexels-photo.jpeg');
    background-size: cover;
}