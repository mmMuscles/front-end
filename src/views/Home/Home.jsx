import './Home.css'
import Auth from "../../components/Auth/Auth";

export default function Home() {
  return (
    <>
      <main className='min-h-screen grid grid-cols-4 grid-rows-3'>
        
        <h1 className='text-white font-black text-5xl row-start-1 row-span-3 col-start-1 col-span-4 bg-black p-5 home-back'>mmmuscles</h1>

        <section className='m-8 p-10 bg-white bg-opacity-80 row-start-1 row-span-2 col-start-3 col-span-2 text-slate-600 flex items-top rounded'>
          <article className=''>
          <p><b>mmmuscles</b> is an awesome and flexible fitness manager, along with a comprehensive and complete list of exercises. Simple to view, simple to use, made by simple people for simple people.</p>
          <br/>
          <p>Create an account today and start your journey to fitness nirvana.</p>
          <br/>
          <p>Mmm, so great!</p>
          </article>
        </section>

        {/* <section className='m-8 p-10 row-start-2 row-span-1 col-start-3 col-span-2 flex flex-col text-white text-sm font-medium justify-center'>
          <span>weekly view</span><img alt='week_view' src='week_view.png' width='250' />
          <span>daily view</span><img alt='day_view' src='day_view.png' width='250'/>
        </section> */}

        <section className='m-8 row-start-3 col-start-1 col-span-4 flex justify-center'>
          <Auth/>
        </section>
        
      </main>
    </>
  );
}
