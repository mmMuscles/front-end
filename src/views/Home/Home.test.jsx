import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { UserProvider } from '../../context/UserContext';
import Home from './Home';

it('should render the home page', () => {
  const { container } = render(
    <UserProvider>
      <MemoryRouter>
        <Home />
      </MemoryRouter>
    </UserProvider>
  );
  expect(container).toMatchSnapshot();
});