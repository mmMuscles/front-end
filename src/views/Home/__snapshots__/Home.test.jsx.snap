// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the home page 1`] = `
<div>
  <main
    class="min-h-screen grid grid-cols-4 grid-rows-3"
  >
    <h1
      class="text-white font-black text-5xl row-start-1 row-span-3 col-start-1 col-span-4 bg-black p-5 home-back"
    >
      mmmuscles
    </h1>
    <section
      class="m-8 p-10 bg-white bg-opacity-80 row-start-1 row-span-2 col-start-3 col-span-2 text-slate-600 flex items-top rounded"
    >
      <article
        class=""
      >
        <p>
          <b>
            mmmuscles
          </b>
           is an awesome and flexible fitness manager. Schedule your day with a simple to use calendar. Select your workouts from a comprehensive and complete list of exercises. Simple to view, simple to use, made by simple people for simple people.
        </p>
        <br />
        <p>
          Create an account today and start your journey to fitness nirvana.
        </p>
        <br />
        <p>
          Mmm, so great!
        </p>
      </article>
    </section>
    <section
      class="m-8 p-10 row-start-2 row-span-1 col-start-3 col-span-2 flex flex-col text-yellow-600 text-sm font-medium justify-center"
    >
      <span>
        daily view
      </span>
      <img
        alt="day_view"
        src="workoutpage.png"
        width="250"
      />
    </section>
    <section
      class="m-8 row-start-3 col-start-1 col-span-4 flex justify-center"
    >
      <div
        class="footie"
      >
        <div
          class="flex"
        >
          <form
            class="flex px-6 mx-6"
          >
            <fieldset>
              <label
                for="email"
              >
                email 
              </label>
              <input
                class="text-black rounded-sm opacity-70 my-1 mr-6 pl-1"
                id="email"
                placeholder="Your email"
                required=""
                type="email"
                value=""
              />
              <label
                for="password"
              >
                password 
              </label>
              <input
                class="text-black rounded-sm opacity-70 my-1 pl-1"
                id="password"
                placeholder="Your password"
                required=""
                type="password"
                value=""
              />
              <button
                class="bg-blue-500 hover:bg-blue-700 text-white font-bold py-1 px-4 mx-6 rounded"
                type="submit"
              >
                SignIn!
              </button>
            </fieldset>
          </form>
           
          <button
            class="bg-slate-700 opacity-90 hover:bg-blue-700 text-white font-bold py-1 px-4 rounded m-15"
          >
            Need to Sign Up?
          </button>
        </div>
      </div>
    </section>
  </main>
</div>
`;
